fc/permission_request_test.go

```diff
@@ -0,0 +1,642 @@
+	multifieldRequest, _ := NewMultifieldRequest(
+		WithNameForMultifieldRequest(&PersonName{Given: "Marian", Family: "Reed"}),
+		WithPlacekeyForMultifieldRequest("226@5z4-zvy-ffz"))
+	err := validateForPermissionFind(multifieldRequest)
+	assert.NoError(t, err)
+}
+
+func TestNewPermissionRequestWithPlacekeyOnly(t *testing.T) {
+	multifieldRequest, _ := NewMultifieldRequest(
+		WithPlacekeyForMultifieldRequest("226@5z4-zvy-ffz"))
+	err := validateForPermissionFind(multifieldRequest)
+	assert.EqualError(t, err, "FullContactError: If you want to use 'location'(or placekey) or 'name' as an input, both must be present and they must have non-blank values")
+}
+
+func TestNewPermissionRequestForCreateWithConsentPurpose(t *testing.T) {
+	consentPurposes := NewConsentPurpose(
+		WithConsentPurposeId(1),
+		WithConsentPurposeChannel("web"),
+		WithConsentPurposeTtl(365),
+		WithConsentPurposeEnabled(true))
+	multifieldRequest, _ := NewMultifieldRequest(
+		WithNameForMultifieldRequest(&PersonName{Given: "Marian", Family: "Reed"}),
+		WithLocationForMultifieldRequest(NewLocation(
+			WithAddressLine1("123/23"),
+			WithPostalCode("23432"))))
+	pr, _ := NewPermissionRequest(
+		WithMultifieldRequestForPermission(multifieldRequest),
+		WithConsentPurposeForPermission(consentPurposes),
+		WithCollectionMethodForPermission("cookiePopUp"),
+		WithCollectionLocationForPermission("Can we get a snapshot of where someone is opting in/out here?"),
+		WithPolicyUrlForPermission("http://foo.baz"),
+		WithTermsServiceForPermission("http://foo.tos"))
+	err := validateForPermissionCreate(pr)
+	assert.NoError(t, err)
+}
+
+func TestNewPermissionRequestForCreateWithMultipleConsentPurpose(t *testing.T) {
+	consentPurpose1 := NewConsentPurpose(
+		WithConsentPurposeId(1),
+		WithConsentPurposeChannel("web"),
+		WithConsentPurposeTtl(365),
+		WithConsentPurposeEnabled(true))
+	consentPurpose2 := NewConsentPurpose(
+		WithConsentPurposeId(2),
+		WithConsentPurposeChannel("phone"),
+		WithConsentPurposeChannel("mobile"),
+		WithConsentPurposeTtl(365),
+		WithConsentPurposeEnabled(true))
+	multifieldRequest, _ := NewMultifieldRequest(
+		WithNameForMultifieldRequest(&PersonName{Given: "Marian", Family: "Reed"}),
+		WithLocationForMultifieldRequest(NewLocation(
+			WithAddressLine1("123/23"),
+			WithPostalCode("23432"))))
+	pr, _ := NewPermissionRequest(
+		WithMultifieldRequestForPermission(multifieldRequest),
+		WithConsentPurposeForPermission(consentPurpose1),
+		WithConsentPurposeForPermission(consentPurpose2),
+		WithCollectionMethodForPermission("cookiePopUp"),
+		WithCollectionLocationForPermission("Can we get a snapshot of where someone is opting in/out here?"),
+		WithPolicyUrlForPermission("http://foo.baz"),
+		WithTermsServiceForPermission("http://foo.tos"))
+	err := validateForPermissionCreate(pr)
+	assert.NoError(t, err)
+}
+
+func TestNewPermissionRequestForCreateWithMultipleConsentPurposeWithoutChannel(t *testing.T) {
```

Comment:
Test Name mentions "WithoutChannel" but channel is present.

Reply:
I'll change